pgoapi/rpc_api.py

```diff
@@ -301,62 +300,96 @@ def _build_main_request(self, subrequests, player_position=None):
-
                 proto_name = entry_name.lower() + '_message'
-                proto_classname = 'pogoprotos.networking.requests.messages.' + proto_name + '_pb2.' + proto_name
-                subrequest_extension = self.get_class(proto_classname)()
-
-                self.log.debug("Subrequest class: %s", proto_classname)
-
-                for key, value in entry_content.items():
-                    if isinstance(value, list):
-                        self.log.debug("Found list: %s - trying as repeated", key)
-                        for i in value:
-                            try:
-                                self.log.debug("%s -> %s", key, i)
-                                r = getattr(subrequest_extension, key)
-                                r.append(i)
-                            except Exception as e:
-                                self.log.warning('Argument %s with value %s unknown inside %s (Exception: %s)', key, i, proto_name, e)
-                    elif isinstance(value, dict):
-                        for k in value.keys():
-                            try:
-                                r = getattr(subrequest_extension, key)
-                                setattr(r, k, value[k])
-                            except Exception as e:
-                                self.log.warning('Argument %s with value %s unknown inside %s (Exception: %s)', key, str(value), proto_name, e)
-                    else:
-                        try:
-                            setattr(subrequest_extension, key, value)
-                        except Exception as e:
-                            try:
-                                self.log.debug("%s -> %s", key, value)
-                                r = getattr(subrequest_extension, key)
-                                r.append(value)
-                            except Exception as e:
-                                self.log.warning('Argument %s with value %s unknown inside %s (Exception: %s)', key, value, proto_name, e)
+                bytes = self._get_proto_bytes('pogoprotos.networking.requests.messages.', proto_name, params)
 
                 subrequest = mainrequest.requests.add()
                 subrequest.request_type = entry_id
-                subrequest.request_message = subrequest_extension.SerializeToString()
+                subrequest.request_message = bytes
 
-            elif isinstance(entry, int):
+            else:
                 subrequest = mainrequest.requests.add()
-                subrequest.request_type = entry
+                subrequest.request_type = entry_id
+
+        return mainrequest
+
+    def _build_platform_requests(self, mainrequest, platform_list):
+        self.log.debug('Generating platform RPC requests...')
+
+        for entry_id, params in platform_list:
+            if params:
+                entry_name = PlatformRequestType.Name(entry_id)
+                if entry_name == 'UNKNOWN_PTR_8':
```

Comment:
The problem is that the code uses the name of the enum value to find the file of the class for the response, every file matches this convention except this message. This can be fixed renaming the class file or the enum name, enum name is the one used for the calls when you want to add it for the request and protobuf class is only used internally that's why I proposed changing class file but it really doesn't matter.

And I haven changed those files as I do not have the protobuf compiler installed/configured  otherwise it would be an fast change.